src/legacy/core_plugins/metrics/public/kbn_vis_types/editor_controller.js

```diff
@@ -45,6 +47,16 @@ function ReactEditorControllerProvider(Private, config) {
         </I18nProvider>, this.el);
     }
 
+    setDefaultIndexPattern = async (appState) => {
```

Comment:
Could you explain what the purpose of this change is? It looks like you're trying to handle the case where the index pattern input is empty, but that probably isn't necessary any more now that we handle null index patterns properly in the KQL serializers.

Reply:
Let me try to check if it can work now without that part.

Reply:
So the reason behind that is the following:
Yes, this was introduced to fix the empty/null pattern error. I'm updating the PR removing that.